### PR TITLE
 Fix an elusive crash in LLDBMemoryReader::readBytes().

### PR DESCRIPTION
**Pull request by adrian-prantl**
Unfortunately I don't have an easy reproducer for this.

<rdar://problem/49973326>

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

File: source/Target/SwiftLanguageRuntime.cpp [hunk 1]
```diff
@@ -1926,14 +1929,17 @@ bool SwiftLanguageRuntime::GetDynamicTypeAndAddress_Protocol(
   lldb::addr_t existential_address;
   bool use_local_buffer = false;
 
-  if (in_value.GetValueType() == eValueTypeConstResult) {
+  if (in_value.GetValueType() == eValueTypeConstResult &&
```
Comment (adrian-prantl) [inline source/Target/SwiftLanguageRuntime.cpp:1932]: This condition alone was just a proxy for what we actually wanted to test, which is in the next line.

Comment (dcci) [reply source/Target/SwiftLanguageRuntime.cpp:1932]: I thought we were going to replace this with `DataExtractor`? Are you doing it as a follow up?

**Comment by dcci** [comment]
```
/home/buildnode/jenkins/workspace/swift-lldb-PR-Linux/branch-swift-5.1-branch/lldb/source/Target/SwiftLanguageRuntime.cpp:1933:45: error: use of undeclared identifier 'eValueTypeHostAddress'
16:05:33       in_value.GetValue().GetValueType() == eValueTypeHostAddress) {
```
This probably needs to be fully qualified.

File: source/Target/SwiftLanguageRuntime.cpp (hunk 1 above)
Comment (adrian-prantl) [reply source/Target/SwiftLanguageRuntime.cpp:1932]: It turned out the the `else` code path is already reading memory from the process.

Comment (dcci) [reply source/Target/SwiftLanguageRuntime.cpp:1932]: Very good, thanks.

**Review by dcci — APPROVED** [review]

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

**Pull request merged**